velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -53,8 +54,77 @@ void WindowFuzzer::addWindowFunctionSignatures(
 }
 
 const std::string WindowFuzzer::generateFrameClause() {
-  // TODO: allow randomly generating the frame clauses.
-  return "RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW";
+  auto frameType = [](int value) -> const std::string {
+    switch (value) {
+      case 0:
+        return "RANGE";
+      case 1:
+        return "ROWS";
+      default:
+        VELOX_UNREACHABLE("Unknown value for frame type generation");
+    }
+  };
+  auto isRowsFrame =
+      boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_);
+  auto frameTypeString = frameType(isRowsFrame);
+
+  constexpr int64_t kMax = std::numeric_limits<int64_t>::max();
+  constexpr int64_t kMin = std::numeric_limits<int64_t>::min();
+  auto frameBound =
+      [&](core::WindowNode::BoundType boundType) -> const std::string {
+    // Generating only constant bounded k PRECEDING/FOLLOWING frames for now.
+    auto kValue =
```

Comment:
I saw the previous version only generates kValue from 1 to 10. Generating kValue between kMin and kMax seems to make it larger than the partition sizes most of the time. Could we make the chances of generating valid values and invalid values (e.g., too large for partitions) more evenly distributed?